_posts/2017-10-24-Bun.md

```diff
@@ -0,0 +1,61 @@
+
+# Fun modules to test: the OCaml standard library
+
+A test framework isn't very useful without something to test.  OCaml 4.06.0 recently entered feature freeze and includes new functions in the standard library, so it seemed reasonable to target some interesting modules in that codebase.
+
+The [ocaml-test-stdlib](https://github.com/yomimono/ocaml-test-stdlib) repository contains some Crowbar tests for the Set and Map modules, including tests adapted from the [unit tests for `Map.update`](https://github.com/ocaml/ocaml/blob/d10f0857a40001f65f4cc00f6ee187cbd5227738/testsuite/tests/basic/maps.ml#L34) added for 4.06.0 .  These tests use several modules from the standard library as inputs for the Make functors of Map and Set.  Int, String, Char, Nativeint, and Uchar are all potential candidates for Map keys or values, as well as Set elements.  Tests for all of these modules (although not all combinations for Map keys and values) are generated, and have an equal chance of being chosen for execution.
+
+## stdlib tests as a proxy for CI
+
+Testing the OCaml standard library in a feature-frozen release isn't a perfect proxy for the workflow we'd like to support in `ocaml-bun`.  Fortunately, continually iterating on the tests themselves is a fairly good proxy for iterating on the code behind them - we have many deploy cycles, and sometimes tests have bugs in them which necessitate retaining and using the crashy input to make improvements.
+
+So far, we've run these tests on a machine with 8 CPU cores, and found no property violations in either beta of 4.06.0.  We ran the same set of tests against the known-buggy 4.04.0, and were pleased to see many property violations in the Set module, so we're confident that at least some classes of property violation are detected.
+
+# Where This Tooling Fits
+
+We can't prove that an implementation is correct with `ocaml-bun`, `afl-fuzz`, or tools built on similar technologies.  Instead, we hope to provide a tool which:
+
+* fits easily into CI workflows
+* requires little developer effort to write useful tests
+* rewards any test effort with outsized likelihood of discovering useful errors
+* delivers results with a minimum of fuss for maximum convenience
+
+We hope that the ease of using `ocaml-bun` and `afl-fuzz` is complementary to approaches which require more developer involvement and attention like verification.
```

Comment:
link to the bun repo, how to try it, should go here?

Reply:
I don't quite want to release it yet, but there's an image on Docker hub that people can pull and try if you want to give some call to action here?

Reply:
that sounds good -- and maybe a note that it'll be released soon "when ready" just to make it clear that it will be open source (just not finished yet for a first release)